mmcv/ops/nms.py

```diff
@@ -90,7 +104,7 @@ def symbolic(g, boxes, scores, iou_threshold, sigma, min_score, method,
 
 
 @deprecated_api_warning({'iou_thr': 'iou_threshold'})
-def nms(boxes, scores, iou_threshold, offset=0):
+def nms(boxes, scores, iou_threshold, score_threshold=0, max_num=-1, offset=0):
```

Comment:
The call of `nms` in Line 306 would be a problem,  because it would reduce `max_num` boxes for each split.

Reply:
I used `mmdetection/tools/test.py` to collect quality metrics and did not notice any difference in `batched_nms` between if branches when using `max_num`.

Reply:
It may have a bad influence for training where there are a large amount of `bboxes`. You could post models and their mertics in the PR description as well.

Reply:
This potential problem can be worked around by moving the changes to `if torch.onnx.is_in_onnx_export()`. Is it okay?

Reply:
It could be a workaround.